frontend/src/metabase/actions/components/ActionViz/Action.unit.spec.tsx

```diff
@@ -0,0 +1,286 @@
+    });
+
+    it("should render a button with custom text", async () => {
+      await setup({
+        settings: { "button.label": "Please Click Me" },
+      });
+      expect(screen.getByRole("button")).toHaveTextContent("Please Click Me");
+    });
+
+    it("clicking an action button should open a modal action form", async () => {
+      await setup();
+
+      userEvent.click(screen.getByRole("button"));
+      expect(screen.getByRole("dialog")).toBeInTheDocument();
+      expect(screen.getByTestId("action-form")).toBeInTheDocument();
+      expect(screen.getByLabelText("Parameter 1")).toBeInTheDocument();
+    });
+
+    it("the modal should have the action name as a title", async () => {
+      await setup();
+
+      userEvent.click(screen.getByRole("button"));
+      expect(screen.getByRole("dialog")).toHaveTextContent("My Awesome Action");
+    });
+
+    it("clicking the cancel button on the form should close the modal", async () => {
+      await setup();
+
+      userEvent.click(screen.getByRole("button"));
+      expect(screen.getByRole("dialog")).toHaveTextContent("My Awesome Action");
+      userEvent.click(screen.getByRole("button", { name: "Cancel" }));
+      expect(screen.queryByRole("dialog")).not.toBeInTheDocument();
+    });
+  });
+
+  describe("Form actions", () => {
+    const formSettings = { actionDisplayType: "form" };
+    it("should render an action form", async () => {
+      await setup({ settings: formSettings });
+
+      expect(screen.getByTestId("action-form")).toBeInTheDocument();
+      expect(screen.getByLabelText("Parameter 1")).toBeInTheDocument();
+    });
+
+    it("should render the action name as the form title", async () => {
+      await setup({ settings: formSettings });
+
+      expect(screen.getByText("My Awesome Action")).toBeInTheDocument();
+    });
+
+    it("should only show form fields with no provided values from dashboard filters", async () => {
+      await setup({
+        settings: formSettings,
+        parameterValues: { "dash-param-1": "foo" },
+      });
+
+      expect(screen.queryByLabelText("Parameter 1")).not.toBeInTheDocument();
+      expect(screen.getByLabelText("Parameter 2")).toBeInTheDocument();
+    });
+
+    it("should render as a button if no parameters are missing", async () => {
+      await setup({
+        settings: formSettings,
+        parameterValues: { "dash-param-1": "foo", "dash-param-2": "bar" },
+      });
+
+      expect(
+        await screen.findByRole("button", { name: "Click me" }),
+      ).toBeInTheDocument();
+    });
+
+    it("should submit provided form input values to the action execution endpoint", async () => {
+      const expectedBody = {
+        modelId: 777,
+        parameters: {
+          "1": "foo",
+          "2": "bar",
+        },
+      };
+
+      const scope = nock(location.origin)
+        .post("/api/dashboard/123/dashcard/456/execute", expectedBody)
+        .reply(200, { "rows-updated": [1] });
```

Comment:
I think it's [preferred](https://www.notion.so/metabase/46-Document-the-pattern-with-setup-function-in-jest-tests-ce3c9f374321427dabe0f26e719aecee) to mock server endpoints inside the `setup`